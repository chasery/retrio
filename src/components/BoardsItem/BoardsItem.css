.BoardsItem__header {
  display: flex;
  flex-wrap: nowrap;
  border-bottom: 0.0625rem solid #ebebeb;
}

.BoardsItem__headerControl {
  display: flex;
  position: relative;
  justify-content: flex-end;
  align-items: center;
  flex-basis: 1rem;
  padding: 1rem;
  font-size: 1.5rem;
  font-weight: bold;
  line-height: 1rem;
}

.BoardsItem__headerInfo {
  min-width: 0;
  flex-grow: 1;
}
.BoardsItem__headerInfo > * {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.BoardsItem__name {
  padding: 1rem 1rem 0;
}

.BoardsItem__date {
  margin: 0;
  padding: 0.25rem 1rem 1rem;
  font-size: 0.75rem;
}
