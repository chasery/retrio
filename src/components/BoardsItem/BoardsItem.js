import React, { useContext, useState } from 'react';
import { Link } from 'react-router-dom';
import RetrioContext from '../../context/retrio-context';
import Menu from '../Menu/Menu';
import BoardsItemThumb from '../BoardsItemThumb/BoardsItemThumb';
import './BoardsItem.css';

function BoardsItem(props) {
  const { id, name, owner_id, created_at, modified_at } = props;
  const context = useContext(RetrioContext);
  const [visible, setVisible] = useState(false);

  const canModify = () => {
    if (owner_id === context.loggedInUser.id) {
      return true;
    } else {
      return false;
    }
  };

  const handleMenuToggle = () => {
    setVisible(!visible);
  };

  const handleDeleteBoard = () => {
    context.deleteCard(id);
  };

  return (
    <li className='BoardsItem'>
      <div className='BoardsItem__header' title={name}>
        <Link className='BoardsItem__headerInfo' to={`/boards/${id}`}>
          <h3 className='BoardsItem__name'>{name}</h3>
          {modified_at ? (
            <p className='BoardsItem__date'>
              Last modified {modified_at.toLocaleDateString()}
            </p>
          ) : (
            <p className='BoardsItem__date'>
              Created {created_at ? created_at.toLocaleDateString() : null}
            </p>
          )}
        </Link>
        {canModify() && (
          <div className='BoardsItem__headerControl'>
            <span onClick={handleMenuToggle}>â‹®</span>
            <Menu visible={visible}>
              <li>
                <Link to={`/boards/${id}/edit-board`}>Edit Board</Link>
              </li>
              <li>
                <button className='Link' onClick={handleDeleteBoard}>
                  Delete Board
                </button>
              </li>
            </Menu>
          </div>
        )}
      </div>
      <Link to={`/boards/${id}`}>
        <BoardsItemThumb />
      </Link>
    </li>
  );
}

export default BoardsItem;
