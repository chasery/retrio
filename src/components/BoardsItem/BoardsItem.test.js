import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BoardsItem from './BoardsItem';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    id: '',
  }),
}));

describe('BoardsItem component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<BoardsItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
