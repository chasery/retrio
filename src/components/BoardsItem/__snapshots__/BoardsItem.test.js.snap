// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardsItem component renders without error 1`] = `
<li
  className="BoardsItem"
>
  <div
    className="BoardsItem__header"
  >
    <Link
      className="BoardsItem__headerInfo"
      to="/boards/undefined"
    >
      <h3
        className="BoardsItem__name"
      />
      <p
        className="BoardsItem__date"
      >
        Created 
      </p>
    </Link>
    <div
      className="BoardsItem__headerControl"
    >
      <span
        onClick={[Function]}
      >
        â‹®
      </span>
      <Menu
        visible={false}
      >
        <li>
          <Link
            to="/boards/undefined/edit-board"
          >
            Edit Board
          </Link>
        </li>
        <li>
          <button
            className="Link"
            onClick={[Function]}
          >
            Delete Board
          </button>
        </li>
      </Menu>
    </div>
  </div>
  <Link
    to="/boards/undefined"
  >
    <BoardsItemThumb />
  </Link>
</li>
`;
